### DISCUSSION TITLE
How about we add "Services.appinfo.invalidateCachesOnRestart(); " to "restartApplication()"

### DISCUSSION THREAD

**Discussion by 117649** [onemen/TabMixPlus#228] [Ideas]
@onemen Add "Services.appinfo.invalidateCachesOnRestart();" should cause FF to clear caches on restart. I think this is what "clear cache" button do when you click it. Hoping this will reduce the chance of TMP not working after install.

**Comment by onemen** [comment]
Where you want to add it?
We don't need to clear cache every time.
And after user need to reinstall the script loader and Tab mix it already too late

**Comment by 117649** [comment]
https://github.com/onemen/TabMixPlus/blob/ef7acb73d7bb80523a36072b7181e5243b8b8dcd/addon/bootstrap.js#L72
Here.

This is where right(and only) after TMP is just installed.
Aren't this is where you would want people to clear cache?

**Reply by onemen** [reply]
When this is runnig Tab mix already installed proprlly.

when user  have cache issue after updating the script loader or firefox they get the message that the app is invalid.

We can not fix this issue with cod in Tab mix.
